### ISSUE TITLE
Gallery relies on Thumbnail Previews

### ISSUE THREAD

**Issue by mattewan** [nextcloud/server#14879]
Nextcloud Gallery displays a preview of a jpeg (or other supported file formats) when clicking on the file.  However, the preview app will "hang" with a spinner until a preview has been generated for the file.

This in itself may be expected behaviour, however nextcloud isn't smart in the way it generates previews.  If you upload 1000 images via the sync client, then browse to that folder via the web interface, scroll all the way to the bottom and click on the last image, the viewer will hang with a spinner until all the previous images in the folder have had previews generated.

I would have expected nextcloud to prioritise preview generation of files that are viewed so that the viewer doesn't hang while nextcloud sequentially generates previews of all the files in the folder.

### Steps to reproduce
1. Upload previewable files to nextcolud via sync client
2. Scroll to bottom of directory via web interface (files here will have thumbnails generated last)
3. Click file.
4. Preview wont load until thumbnails for all previous images have completed

### Expected behaviour
1. Upload previewable files to nextcloud via sync client
2. Scroll to bottom of directory via web interface
3. Click file.
4. Generate preview on demand (don't wait for all previous images to complete)

### Server configuration

**Operating system**: Gentoo

**Web server:** Apache 

**Database:** MySQL

**PHP version:** Php 7.2

**Nextcloud version:** (see Nextcloud admin page) 14.0.6

**Comment by Nils160988** [comment]
More a feature request than a bug I think, but not stale...

**Comment by dgtlmoon** [comment]
I agree, thumbnails seem to be a complete deal breaker for many installs, it makes the entire stack feel like a steamy pile, beginners are even ending up googling how to fix the thumbnails

when they could just push their data to google or apple and it just works... for viewing thumbnails, which humans have been doing on computers for decades now

https://ownyourbits.com/2019/06/29/understanding-and-improving-nextcloud-previews/

**Comment by dgtlmoon** [comment]
yeah i see it, i'm at the bottom of the list and i'm just waiting for something... anything

**Comment by szaimen** [comment]
There are two workarounds for this:
1. install and execute the previewgenerator to generate your previews before they are needed 
2. buy a powerful server that can handle the preview generation on the fly fast enough so that it doesn't take as much time to generate them.

---

But unfortunately the described feature is not really high on our priority list. But this doesn't mean that we don't like the idea or that we are against this feature per se. Quite the opposite! 🙂
There are many way to get new features in Nextcloud:
1. The easiest and most straight forward way (especially here at Github): pull request.
Nextcloud is completely Free Software, this means that everyone is welcome to join and to contribute. If you or anyone else want to work on this feature, this would be great! We appreciate every pull request, and we are definitely able to help in case of questions, reviews, etc.
2. We have a category for Freelancers in our form. Another option would be to post an offer there and try to find a freelancer who want to work on it.
3. The most direct way for a company or organization to get the issue addressed is to get a Enterprise Subscription. This includes everything to enable you to run Nextcloud in a productive environment with guaranteed SLA's and more. The Enterprise Subscription also includes optional professional services such as custom development. Feel free to reach out to us. We are happy to explore the possibilities how to make Nextcloud fit your needs.

**Comment by nachoparker** [comment]
no offense, but I sent a PR that had 0 traction

**Comment by szaimen** [comment]
I suppose in this case I was wrong closing this issue. But does it change something?

**Comment by szaimen** [comment]
PR is here: https://github.com/nextcloud/server/pull/16158

**Comment by darmbrust** [comment]
Here is a wild thought.
How about Gallery just stops sending preview images when you click on them when they don't exist yet?  Just send the file.

The fact that we still can't configure nextcloud to just stop all of the preview nonsense - make small thumbs for grid views - and send original images for everything else - its really quite mind boggling.

Quite honestly, I think nextcloud should just throw it all away, and integrate another 3rd party photo tool as an app in nextcloud that actually works, and includes modern photo viewer features.  Spending this many years trying (and failing, in my view) to even get thumbnails right... Its just silly at this point.  There are entire other teams out there working to do photos right:

https://github.com/piwigo/piwigo
https://github.com/photoprism/photoprism

Or dozens and dozens of opensource javascript photo viewers, that just need photos fed into them.

And I'm sure many more that I haven't heard of.  If the APIs are inadequate to allow that sort of tool integration, it seems that a better use of time would be to improve the integration APIs, rather than continuing to waste time reinventing the wheel on a photo app.

**Comment by szaimen** [comment]
Unfortunately integrating another open source project for photo viewing wouldn't solve this issue as previews are deeply integrated into Nextcloud itself...

**Comment by skjnldsv** [comment]
With the new Files, it won't wait for the entire files array to load, it will cancel previews fetching on items that aren't visible on the page.
Opening the Viewer will take priority on the preview generation :)
